packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -1202,6 +1202,7 @@ function pushAttribute(
       pushStyleAttribute(target, value);
       return;
     }
+    case 'data':
```

Comment:
I'd just follow `pushStartAnchor`'s lead and make a specific `pushStartObject` method that does the sanitization specifically for that tag.